kcp/run.sh

```diff
@@ -0,0 +1,201 @@
+#!/bin/bash
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "$0")" >/dev/null; pwd)"
+
+KUBECONFIG_KCP="${KUBECONFIG_KCP:-$HOME/.kube/kcp.config}"
+KUBECONFIG_KCP_ARGOCD="${KUBECONFIG_KCP_ARGOCD:-$HOME/.kube/kcp.argocd-manager.config}"
+KUBECONFIG_PLNSVC_CLUSTERADMIN="${KUBECONFIG_PLNSVC_CLUSTERADMIN:-$HOME/.kube/plnscv.clusteradmin.config}"
+KUBECONFIG_MERGED="merged-config.kubeconfig:$KUBECONFIG_PLNSVC_CLUSTERADMIN:$KUBECONFIG_KCP:$KUBECONFIG_KCP_ARGOCD"
+
+
+
+usage(){
+  echo "
+Usage:
+    ${0##*/} [options]
+
+Setup the pipeline service on a cluster running on KCP.
+
+Optional arguments:
+    -k, --kcp KCP_CONTEXT
+        Select the KCP instance on which to deploy. Must be one of [stable, unstable].
+        Default value: unstable
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+
+Example:
+    ${0##*/} --all
+" >&2
+}
+
+
+
+parse_args(){
+  KCP_ENV="kcp-unstable"
+
+  local args
+  args="$(getopt -o dhk: -l "debug,help,kcp" -n "$0" -- "$@")"
+  eval set -- "$args";
+  while true; do
+    case $1 in
+      -k|--kcp)
+        shift
+        case $1 in
+          stable|unstable)
+            KCP_ENV="kcp-$1"
+            ;;
+          *)
+            echo "Unsupported environment: $1" >&2
+            echo "" >&2
+            usage
+            exit 1
+            ;;
+        esac
+        ;;
+      -d|--debug)
+        set -x
+        ;;
+      -h|--help)
+        usage
+        exit 0
+        ;;
+      --)
+        # End of arguments
+        break
+      ;;
+      *)
+        echo "Unknown argument: $1"
+        usage
+        exit 1
+        ;;
+    esac
+    shift
+  done
+}
+
+
+
+configure_pipeline_cluster(){
+  echo "[Pipeline cluster]"
+  export KUBECONFIG="$KUBECONFIG_MERGED"
+
+  echo -n "  - Register to KCP: "
+  local cluster_name="pipeline-cluster"
+  local manifests_dir="$SCRIPT_DIR/work/manifests"
+  if ! KUBECONFIG=$KUBECONFIG_KCP oc get cluster "$cluster_name" >/dev/null 2>&1; then
```

Comment:
In the kcp context, do we still have a CRD called 'cluster'? I only see workloadcluster listed under api-resources.

(although, I have not run this script to check if its working as I could not get to this point)

Reply:
Good catch!  The code was always recreating the resource, so I did not see it was broken.